c/src/core/codec.c

```diff
@@ -655,7 +693,7 @@ int pn_data_vfill(pn_data_t *data, const char *fmt, va_list ap)
         if (parent->atom.type == PN_ARRAY) {
           parent->type = (pn_type_t) va_arg(ap, int);
         } else {
-          return pn_error_format(data->error, PN_ERR, "naked type");
+          return pn_error_format(pn_data_error(data), PN_ERR, "naked type");
```

Comment:
Small nit: these should all probably explicitly be the inlined version now `pni_data_error`.